### PR TITLE
Tractor beam QOL

### PR DESCRIPTION
**Pull request by SciathGH**

### CODE REVIEW

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import net.starlegacy.util.isWallSign
+import org.bukkit.Material
+import org.bukkit.block.BlockFace
+import org.bukkit.block.Sign
+import org.bukkit.event.EventHandler
+import org.bukkit.event.player.PlayerToggleSneakEvent
+
+object PlayerShiftListener : SLEventListener() {
+	@EventHandler
+	fun onPlayerInteractEventB(event: PlayerToggleSneakEvent) {
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt:17]:
```suggestion
	fun onPlayerToggleSneakEvent(event: PlayerToggleSneakEvent) {
```

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import net.starlegacy.util.isStainedGlass
+import net.starlegacy.util.isWallSign
+import org.bukkit.Material
+import org.bukkit.block.BlockFace
+import org.bukkit.block.Sign
+import org.bukkit.event.EventHandler
+import org.bukkit.event.player.PlayerToggleSneakEvent
+
+object PlayerShiftListener : SLEventListener() {
+	@EventHandler
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt:16]:
```suggestion
	@EventHandler(priority = EventPriority.MONITOR)
```

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerJumpListener.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import net.starlegacy.util.isStainedGlass
+import net.starlegacy.util.isWallSign
+import org.bukkit.Material
+import org.bukkit.block.BlockFace
+import org.bukkit.block.Sign
+import org.bukkit.event.EventHandler
+import org.bukkit.event.EventPriority
+
+object PlayerJumpListener : SLEventListener() {
+	@EventHandler(priority = EventPriority.LOWEST)
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerJumpListener.kt:17]:
```suggestion
	@EventHandler(priority = EventPriority.MONITOR)
```

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerJumpListener.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	fun onPlayerToggleSneakEvent(event: PlayerJumpEvent) {
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerJumpListener.kt:18]:
```suggestion
	fun onPlayerJumpEvent(event: PlayerJumpEvent) {
```

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import net.starlegacy.util.LegacyBlockUtils
+import net.starlegacy.util.isStainedGlass
+import net.starlegacy.util.isWallSign
+import org.bukkit.Material
+import org.bukkit.block.BlockFace
+import org.bukkit.block.Sign
+import org.bukkit.event.EventHandler
+import org.bukkit.event.player.PlayerToggleSneakEvent
+
+object PlayerShiftListener : SLEventListener() {
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt:15]:
```suggestion
object PlayerToggleSneakListener : SLEventListener() {
```

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+				}
+
+				if (distance < 3) return
+
+				val relative = below.getRelative(BlockFace.DOWN, distance)
+				if (relative.type != Material.AIR) {
+					event.player.teleport(relative.location.add(0.5, 1.5, 0.5))
+					}
+				}
+			}
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt:47]: Misaligned brackets

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt (hunk 6 above)
Comment (SciathGH) [reply src/main/kotlin/net/horizonsend/ion/core/listeners/PlayerShiftListener.kt:47]: thats ktlints fault

**Pull request merged**